### ISSUE TITLE
JSON Schema validation error without mandatory security vocabulary term with default

### ISSUE THREAD

**Issue by ToruKawaguchi** [thingweb/thingweb-playground#34]
JSON Schema validation error occurs at securityDefinitions, when mandatory vocabularies defined at <https://w3c.github.io/wot-thing-description/#sec-security-vocabulary-definition> is omitted, even if the default value is defined at <https://w3c.github.io/wot-thing-description/#json-serializiation-section>.

For example, following fails: 

```jsonld
  "securityDefinitions": {"bearer_sc": {
    "scheme": "bearer",
    "format": "jwt",
    "alg": "ES256",
    "authorization": "https://w3c.p-wot.com:8443/auth"
  }},
  "security": ["bearer_sc"],

```

with error message: 

```
> data.securityDefinitions['bearer_sc'].scheme should be equal to one of the allowed values, data.securityDefinitions['bearer_sc'].scheme should be equal to one of the allowed values, data.securityDefinitions['bearer_sc'].scheme should be equal to one of the allowed values, data.securityDefinitions['bearer_sc'].scheme should be equal to one of the allowed values, data.securityDefinitions['bearer_sc'] should have required property 'in', data.securityDefinitions['bearer_sc'].scheme should be equal to one of the allowed values, data.securityDefinitions['bearer_sc'].scheme should be equal to one of the allowed values, data.securityDefinitions['bearer_sc'].scheme should be equal to one of the allowed values, data.securityDefinitions['bearer_sc'].scheme should be equal to one of the allowed values, data.securityDefinitions['bearer_sc'].scheme should be equal to one of the allowed values, data.securityDefinitions['bearer_sc'] should match exactly one schema in oneOf
```

while following passes: 

```jsonld
  "securityDefinitions": {"bearer_sc": {
    "scheme": "bearer",
    "in": "header", 
    "format": "jwt",
    "alg": "ES256",
    "authorization": "https://w3c.p-wot.com:8443/auth"
  }},
  "security": ["bearer_sc"],

```

**Comment by egekorkan** [comment]
First of all, sorry about the error message, it happens because of the validation library which puts an error for every possible securityDefinition schema that fails (in this case there are 10 or so possibilities).
Regarding the default values: It was not clear to me that there is more, I will correct this.

**Comment by egekorkan** [comment]
What do you think of the following validation:
- A simple schema that outputs "validated" for a simple and full TD
- A full schema that outputs "validated" only for a full TD

Both would have their own lights in Playground

**Comment by ToruKawaguchi** [comment]
Thanks! For now, as testing purpose during spec creation, that would be helpful.

When we finalize the spec, we'd better to be careful about that, because if it looks like there's two different version of TD spec then readers would be confused.

**Comment by egekorkan** [comment]
Alright, working on it. Will finish by today evening!

**Comment by egekorkan** [comment]
Fixed with #35 , can you verify and if fixed close the issue?

**Comment by ToruKawaguchi** [comment]
Thanks. Worked well.
